### PR TITLE
Add new extractor for Dagelijkse kost

### PR DESCRIPTION
**Pull request by paretje**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [X] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [X] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [X] Covered the code with tests (note that PRs without tests will be REJECTED)
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [X] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This PR adds a new extractor for "Dagelijkse kost" to `canvas.py`. The extractor is based on the existing extractor for een and canvas.

### CODE REVIEW

File: youtube_dl/extractor/canvas.py [hunk 1]
```diff
@@ -332,3 +332,47 @@ def _real_extract(self, url):
...
+            group='id')
+
+        return {
+            '_type': 'url_transparent',
+            'url': 'https://mediazone.vrt.be/api/v1/dako/assets/%s' % (video_id),
+            'ie_key': CanvasIE.ie_key(),
+            'id': video_id,
+            'display_id': display_id,
+            'title': title,
+            'description': self._og_search_description(webpage),
```
Comment (remitamine) [inline youtube_dl/extractor/canvas.py:377]: fallback to other available values.

File: youtube_dl/extractor/canvas.py [hunk 2]
```diff
@@ -332,3 +332,47 @@ def _real_extract(self, url):
             'display_id': display_id,
             'season_number': int_or_none(page.get('episode_season')),
         })
+
+
+class DagelijkseKostIE(InfoExtractor):
+    IE_DESC = 'dagelijksekost.een.be'
+    _VALID_URL = r'https?://dagelijksekost\.een\.be/(?:[^/]+/)*(?P<id>[^/?#&]+)'
```
Comment (remitamine) [inline youtube_dl/extractor/canvas.py:339]: match only supported URLs.

File: youtube_dl/extractor/canvas.py [hunk 3]
```diff
@@ -332,3 +332,47 @@ def _real_extract(self, url):
...
+            webpage, 'title', default=None) or self._og_search_title(
+            webpage, default=None))
+
+        video_id = self._html_search_regex(
+            r'data-url=(["\'])(?P<id>(?:(?!\1).)+)\1', webpage, 'video id',
+            group='id')
+
+        return {
+            '_type': 'url_transparent',
+            'url': 'https://mediazone.vrt.be/api/v1/dako/assets/%s' % (video_id),
```
Comment (remitamine) [inline youtube_dl/extractor/canvas.py:372]: parentheses not needed.

**Review by remitamine — CHANGES_REQUESTED** [review]

File: youtube_dl/extractor/canvas.py [hunk 4]
```diff
@@ -15,11 +15,12 @@
     merge_dicts,
     str_or_none,
     url_or_none,
+    get_element_by_class,
```
Comment (remitamine) [inline youtube_dl/extractor/canvas.py:18]: Alphabetic order.

File: youtube_dl/extractor/canvas.py [hunk 5]
```diff
@@ -332,3 +333,50 @@ def _real_extract(self, url):
...
+
+        title = strip_or_none(get_element_by_class(
+            "dish-metadata__title",
+            webpage) or self._og_search_title(
+            webpage, default=None))
+
+        description = strip_or_none(get_element_by_class(
+            "dish-description",
+            webpage) or self._og_search_description(
+            webpage, default=None))
```
Comment (remitamine) [inline youtube_dl/extractor/canvas.py:368]: `twitter:description` and `description` meta tags are also available.

File: youtube_dl/extractor/canvas.py [hunk 6]
```diff
@@ -332,3 +333,52 @@ def _real_extract(self, url):
...
+    }
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+
+        title = strip_or_none(get_element_by_class(
+            'dish-metadata__title',
+            webpage) or self._og_search_title(
+            webpage, default=None))
```
Comment (remitamine) [inline youtube_dl/extractor/canvas.py:363]: check the result of the fallback(in comparision with the primary source).

File: youtube_dl/extractor/canvas.py [hunk 7]
```diff
@@ -332,3 +333,52 @@ def _real_extract(self, url):
...
+
+        description = strip_or_none(
+            get_element_by_class(
+                'dish-description', webpage)
+            or self._html_search_meta(
+                ('description', 'twitter:description'), webpage)
+            or self._og_search_description(webpage, default=None))
```
Comment (remitamine) [inline youtube_dl/extractor/canvas.py:370]: combine into a single call to `_html_search_meta`.

File: youtube_dl/extractor/canvas.py [hunk 8]
```diff
@@ -332,3 +333,52 @@ def _real_extract(self, url):
...
+        webpage = self._download_webpage(url, display_id)
+
+        title = strip_or_none(get_element_by_class(
+            'dish-metadata__title',
+            webpage) or self._og_search_title(
+            webpage, default=None))
+
+        description = strip_or_none(
+            get_element_by_class(
+                'dish-description', webpage)
```
Comment (remitamine) [inline youtube_dl/extractor/canvas.py:367]: check that extracted description is what is expected(should not contain html tags).

**Review by remitamine — CHANGES_REQUESTED** [review]

File: youtube_dl/extractor/canvas.py (hunk 6 above)
Comment (paretje) [reply youtube_dl/extractor/canvas.py:364]: Should I drop the fallback, or apply a regex to make up for the differences?

**Review by remitamine — CHANGES_REQUESTED** [review]

File: youtube_dl/extractor/canvas.py [hunk 9]
```diff
@@ -332,3 +334,54 @@ def _real_extract(self, url):
...
+            webpage, default=None))
+
+        description = strip_or_none(
+
+            clean_html(get_element_by_class(
+                'dish-description', webpage))
+            or self._html_search_meta(
+                ('description', 'twitter:description', 'og:description'),
+                webpage,
+                default=None))
```
[the remark this answers is not in the document]
Comment (paretje) [reply youtube_dl/extractor/canvas.py]: Should I do the same for title?

[the remark this answers is not in the document]
Comment (remitamine) [reply youtube_dl/extractor/canvas.py]: yes.

**Pull request merged**